src/uu/sort/src/sort.rs

```diff
@@ -322,25 +322,50 @@ struct Precomputed {
 
 impl GlobalSettings {
     /// Parse a SIZE string into a number of bytes.
-    /// A size string comprises an integer and an optional unit.
+    /// A size string comprises an integer and an optional unit or percentage of total memory.
```

Comment:
Do you know whether all uses of a size should accept this? If it's just `sort` we might have to move this edge case somewhere else.

Reply:
I did it with intention of only making it work for sort. So, I didn't do it in Parser file. And, isn't this Global Settings for sort?

Reply:
Oh yeah you're right! My mistake :)

Reply:
this is impl in Global settings. If you want, i can take it to buffer settings option where it will check for % before sending to parsing